main

```diff
@@ -0,0 +1,347 @@
+ }
+ }
+}
+}
+void backButton(){
+ float xDist = 60;
+ float yDist = 20;
+ if (abs(mouseX - 120) <= xDist && abs(mouseY - 40) <= yDist){
+ background(0);
+ fill(100); 
+ rect(75,20,120,40,10);
+ fill(200);
+ textSize(38);
+ text("back", 95, 52);
+ } else {
+ background(0);
+ fill(50);
+ rect(75,20,120,40,10);
+ fill(100);
+ textSize(38);
+ text("back", 95, 52);
+ }
+}
+
+// speed changes of the partical of the colliding with other object
+void collide(){
+if (collisions == true) {
+if (collision.size() > 1) {
+for (int i = 0; i < collision.size() - 1; i ++) {
+ for (int j = i + 1; j < collision.size(); j ++) {
+ float dx = collision.get(i).x - collision.get(j).x;
+ float dy = collision.get(i).y - collision.get(j).y;
+ float distance = abs(sqrt(dx*dx + dy*dy));
+ float bump = collision.get(i).size/2 + collision.get(j).size/2;
+ if (distance <= bump) {
+ float theta = atan2(dy, dx);
+ float endX = collision.get(i).x + cos(theta)*bump;
+ float endY = collision.get(i).y + sin(theta)*bump;
+ collision.get(i).vx = (endX - collision.get(j).x)/(collision.get(i).size);
+ collision.get(i).vy = (endY - collision.get(j).y)/(collision.get(i).size);
+ collision.get(j).vx = (collision.get(j).x - endX)/(collision.get(j).size);
+ collision.get(j).vy = (collision.get(j).y - endY)/(collision.get(j).size);
+ }
+ }
+
+ }
+ }
+ }
+if (interactiveCollision == true) {
+if (interactiveCollisions.size() > 1) {
+for (int i = 0; i < interactiveCollisions.size() - 1; i ++) {
+ for (int j = i + 1; j < interactiveCollisions.size(); j ++) { 
+ float dx = interactiveCollisions.get(i).x - interactiveCollisions.get(j).x;
+ float dy = interactiveCollisions.get(i).y - interactiveCollisions.get(j).y;
+ float distance = abs(sqrt(dx*dx + dy*dy));
+ float bump = interactiveCollisions.get(i).size/2 + interactiveCollisions.get(j).size/2;
+ if (distance <= bump) {
+ float theta = atan2(dy, dx);
+ float endX = interactiveCollisions.get(i).x + cos(theta)*bump;
+ float endY = interactiveCollisions.get(i).y + sin(theta)*bump;
+ interactiveCollisions.get(i).vx = (endX - interactiveCollisions.get(j).x)/(interactiveCollisions.get(i).size);
+ interactiveCollisions.get(i).vy = (endY - interactiveCollisions.get(j).y)/(interactiveCollisions.get(i).size);
+ interactiveCollisions.get(j).vx = (interactiveCollisions.get(j).x - endX)/(interactiveCollisions.get(j).size);
+ interactiveCollisions.get(j).vy = (interactiveCollisions.get(j).y - endY)/(interactiveCollisions.get(j).size);
+ }
+ }
+
+ }
+ }
+ }
+}
+
+// collision of the particles
+class collisionParticles{
+float x;
+float y;
+float vx;
+float vy;
+float size;
+float r;
+float g;
+float b;
+
+ collisionParticles(float ix, float iy, float ivx, float ivy, float isize, float ir, float ig, float ib) {
+ x = ix;
+ y = iy;
+ vx = ivx;
+ vy = ivy;
+ size = isize;
+ r = ir;
+ g = ig;
+ b = ib; 
+ }
+
+void update() {
+ if (vx > 0){
+ if (x >= width-1) {
+ if (vx > 0){
+ vx = -vx;
+ }
+ }
+ if (x <=0) {
+ if (vx < 0) {
+ vx = vx;
```

Comment:
Yeah, those lines don't really do anything. The reason why I leave it there because I want to make sure that the variable is assigned to the same positive value and also I don't confuse with if (x <=0) {
        if (vx < 0) {
          **vx = -vx;**
        }
      }